src/Hintman/HLint.hs

```diff
@@ -55,11 +56,14 @@ getFileHLintComments
     => FileDeltas
     -> (FilePath, Maybe ByteString)
     -> m [Comment]
-getFileHLintComments fd (fileName, maybeContent) = case maybeContent of
+getFileHLintComments fds (fileName, maybeContent) = case maybeContent of
     Nothing -> [] <$ log I ("Content is empty: " <> toText fileName)
     Just (decodeUtf8 -> content) -> do
         ideas <- getFileHLintSuggestions fileName content
-        pure $ mapMaybe (createComment fd fileName content) ideas
+        let fileDelta = find ((==) (toText fileName) . fileDeltaDestFile) fds
```

Comment:
Looks like this in this code block we need `fileName` as both `String` and `Text`. What do you think about creating a variable like `fileNameText :: Text`? Then we can use it instead of `toText fileName`. And we also can simplify this `find` like this:

```haskell
let fileDelta = find (\FileDelta{..} -> fileDeltaDestFile == fileNameText) fds
```

I'm not sure about the best solution here, I always find it awkward to work with the same variable when it is required multiple times under different string representations.